backend/src/main/scala/cromwell/backend/WorkflowBackendActor.scala

```diff
@@ -0,0 +1,83 @@
+package cromwell.backend
+
+import akka.actor.{ActorRef, ActorLogging, Actor}
+import akka.event.LoggingReceive
+import cromwell.backend.WorkflowBackendActor._
+import cromwell.core.CallOutputs
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+object WorkflowBackendActor {
+
+  // Commands
+  sealed trait WorkflowBackendActorMessage
+
+  case class Execute(jobDescriptor: BackendJobDescriptor) extends WorkflowBackendActorMessage
+  case class Recover(jobDescriptor: BackendJobDescriptor) extends WorkflowBackendActorMessage
+  case class Abort(jobDescriptor: BackendJobDescriptor) extends WorkflowBackendActorMessage
+
+  // Events
+  sealed trait WorkflowBackendActorResponse
+
+  sealed trait ExecutionResponse extends WorkflowBackendActorResponse
+  case class ExecutionSucceeded(jobDescriptor: BackendJobDescriptor, callOutputs: CallOutputs) extends ExecutionResponse
+  case class ExecutionFailed(jobDescriptor: BackendJobDescriptor, throwable: Throwable) extends ExecutionResponse
+  case class ExecutionFailedRetryable(jobDescriptor: BackendJobDescriptor, throwable: Throwable) extends ExecutionResponse
+
+  sealed trait AbortResponse extends WorkflowBackendActorResponse
+  case class AbortSucceeded(jobDescriptor: BackendJobDescriptor) extends AbortResponse
+  case class AbortFailed(jobDescriptor: BackendJobDescriptor, throwable: Throwable) extends AbortResponse
+}
+
+/**
+  * Workflow-level actor for executing, recovering and aborting jobs.
+  */
+trait WorkflowBackendActor extends Actor with ActorLogging {
+
+  implicit def ec: ExecutionContext
+  def workflowDescriptor: BackendWorkflowDescriptor
```

Comment:
Eventually yes but that should probably be left to whoever is the lucky recipient of that ticket.